### PR TITLE
added point loads and their IO

### PR DESCRIPTION
**Pull request by morettid**
implementation of point load related to issue #13

### CODE REVIEW

File: stem/kratos_IO.py [hunk 1]
```diff
@@ -47,16 +52,78 @@ def __write_output_processes(self):
...
+        load_dict: Dict[str, Any] = {
+            "python_module": None,
+            "kratos_module": None,
+            "process_name": None,
+            "Parameters": {},
+        }
+
+        # add load parameters to dictionary based on load type.
+        if isinstance(load.load_parameters, PointLoad):
+            load_dict.update(
```
Comment (aronnoordam) [inline stem/kratos_IO.py:79]: can you make a seperate function for this e.g. __create_point_load_dict, such that the code stays a bit more readable (for when multiple types of loads will be added in the future

Comment (aronnoordam) [inline stem/kratos_IO.py:79]: dictionary.update, is only required if you extend the dictionary with new keys. In this case you can do load_dict["python_module"] =  "apply_vector_constraint_table_process"

File: stem/kratos_IO.py [hunk 2]
```diff
@@ -47,16 +52,78 @@ def __write_output_processes(self):
...
+                {
+                    "python_module": "apply_vector_constraint_table_process",
+                    "kratos_module": "KratosMultiphysics.GeoMechanicsApplication",
+                    "process_name": "ApplyVectorConstraintTableProcess",
+                }
+            )
+            load_dict.update(
```
Comment (aronnoordam) [inline stem/kratos_IO.py:86]: same

File: stem/kratos_IO.py [hunk 3]
```diff
@@ -47,16 +52,78 @@ def __write_output_processes(self):
...
 
+    def __write_constraints(self):
+        pass
+
+    def __write_loads(self):
+        pass
+
+    def __create_load_dict(self, load: Load) -> Dict[str, Any]:
+        """
+        Creates a dictionary containing the load parameters
```
Comment (aronnoordam) [inline stem/kratos_IO.py:63]: can you add extra Enters between description; Args; Returns (Sorry)

File: stem/load.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    Abstract base class for load parameters
+    """
+    pass
+
+
+@dataclass
+class PointLoad(LoadParametersABC):
+    """
+    Class containing the load parameters for a point load
+    :Attributes:
```
Comment (aronnoordam) [inline stem/load.py:18]: Enter

Comment (aronnoordam) [inline stem/load.py:18]: Attributes: instead of  ":Attributes:"

File: stem/kratos_IO.py [hunk 5]
```diff
@@ -1,4 +1,8 @@
+import json
+
 import numpy as np
+from stem.load import Load, PointLoad
+from typing import List, Dict, Any
```
Comment (aronnoordam) [inline stem/kratos_IO.py:5]: order of inputs should be: python imports -> external packages -> project imports, with Enters between the different types

File: stem/load.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    """
+
+    active: List[bool] = field(default_factory=lambda: [True, True, True])
+    value: List[float] = field(default_factory=lambda: [0.0, 0.0, 0.0])
+
+
+class Load:
+    """
+    Class containing load information acting on a body part
+    Attributes:
```
Comment (aronnoordam) [inline stem/load.py:30]: enter

File: stem/load.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        name (str): name of the load
+        load_parameters (LoadParametersABC): class containing load parameters
+    """
+
+    def __init__(self, name: str, load_parameters: LoadParametersABC):
+        """
+        Constructor of the load class
+        Args:
```
Comment (aronnoordam) [inline stem/load.py:38]: enter

**Review by aronnoordam — COMMENTED** [review]
minor style remarks

File: stem/kratos_IO.py [hunk 8]
```diff
@@ -47,16 +52,84 @@ def __write_output_processes(self):
...
+    def __write_loads(self):
+        pass
+
+    @staticmethod
+    def __create_point_load_dict(load: Load) -> Dict[str, Any]:
+        """
+        Creates a dictionary containing the point load parameters
+
+        Args:
+            load (PointLoad): point load object
```
Comment (aronnoordam) [inline stem/kratos_IO.py:67]: type is not correct

File: stem/kratos_IO.py [hunk 9]
```diff
@@ -47,16 +52,84 @@ def __write_output_processes(self):
...
+        else:
+            raise NotImplementedError
+
+    def create_loads_process_dictionary(self, loads: List[Load]) -> Dict[str, Any]:
+        """
+        Creates a dictionary containing the load_process_list (list of
+        dictionaries to specify the loads for the model)
+
+        Args:
+            loads (List[Material]): list of load objects
```
Comment (aronnoordam) [inline stem/kratos_IO.py:110]: type is not correct

File: tests/utils.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+import numpy.testing as npt
+
+
+class TestUtils:
+
+    @staticmethod
+    def assert_dictionary_almost_equal(expected: Dict[Any, Any], actual: Dict[Any, Any]):
+        """
+        Checks whether two dictionaries are equal.
```
Comment (aronnoordam) [inline tests/utils.py:11]:
```suggestion
        Checks whether two dictionaries are equal.
        
```

File: stem/kratos_IO.py (hunk 5 above)
Comment (aronnoordam) [reply stem/kratos_IO.py:5]: i.e.: json, typing -> numpy -> stem

Comment (morettid) [reply stem/kratos_IO.py:5]: Indeed, forgot to run the import optimizer!

**Review by aronnoordam — APPROVED** [review]

**Pull request merged**